src/py-opentimelineio/opentimelineio/adapters/otiod.py

```diff
@@ -0,0 +1,149 @@
+#
+# Copyright 2019 Pixar Animation Studios
+#
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+
+"""OTIOD adapter - bundles otio files linked to local media in a directory
+
+Takes as input an OTIO file that has media references which are all
+ExternalReferences with target_urls to files with unique basenames that are
+accessible through the file system and bundles those files and the otio file
+into a single directory named with a suffix of .otiod.
+"""
+
+import os
+import copy
+import shutil
+
+from . import (
+    file_bundle_utils as utils,
+    otio_json,
+)
+
+from .. import (
+    exceptions,
+)
+
+try:
+    # Python 2.7
+    import urlparse
+except ImportError:
+    # Python 3
+    import urllib.parse as urlparse
+
+
+def read_from_file(filepath, absolute_media_reference_paths=False):
+    result = otio_json.read_from_file(
+        os.path.join(filepath, utils.BUNDLE_PLAYLIST_PATH)
+    )
+
+    if not absolute_media_reference_paths:
+        return result
+
+    for cl in result.each_clip():
+        try:
+            source_fpath = cl.media_reference.target_url
+        except AttributeError:
+            continue
+
+        rel_path = urlparse.urlparse(source_fpath).path
+        new_fpath = "file:{}".format(os.path.join(filepath, rel_path))
```

Comment:
In light of some recent discussions we've had about what the form of `target_url` should be, since these are built as relative URLs, maybe we can drop the `file:` off the front to match the types of relative URLs you'd use in HTML.
so instead of:
`file:media/MyMovieFile.mp4`

It comes out:
`media/MyMovieFile.mp4`